Gemfile

```diff
@@ -7,8 +7,12 @@ gem "manageiq-loggers", "~> 0.1.0"
 gem "manageiq-messaging"
 gem "optimist"
 
-gem "sources-api-client",         :git => "https://github.com/ManageIQ/sources-api-client-ruby", :branch => "master"
-gem "topological_inventory-core", :git => "https://github.com/ManageIQ/topological_inventory-core", :branch => "master"
+gem "sources-api-client",                       :git => "https://github.com/ManageIQ/sources-api-client-ruby", :branch => "master"
+gem "topological_inventory-core",               :git => "https://github.com/ManageIQ/topological_inventory-core", :branch => "master"
+gem "topological_inventory-ingress_api-client", :git => "https://github.com/ManageIQ/topological_inventory-ingress_api-client-ruby", :branch => "master"
+
+gem "typhoeus"
+gem 'http'
```

Comment:
@slemrmartin if you rebase this PR you'll have the http gem already, https://github.com/ManageIQ/topological_inventory-sync/blob/master/Gemfile#L7

Reply:
ah, ok. 


> @slemrmartin I'm extracting #8 so that you can use it in this PR, so duplicating the whole method doesn't make a lot of sense

That would be great! Please look at this variation, because HTTP.get doesn't work with insights-upload svc with localstorage settings

Reply:
Oh it looks like there is a bug with the upload service, it returns `url: /tmp/uploads/...` to the validator but `url: file:///tmp/uploads/...` to the processor worker.  I'll put in a PR to fix that but for now I'll handle both `nil` and `file` schemes here rather than have to support File, typhoeus, and http access methods.